utils/validators/validators_test.go

```diff
@@ -0,0 +1,166 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package validators contains environment variables validator.
+package validators
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+)
+
+func TestEnvVarValidator(t *testing.T) {
+	type args struct {
+		env []string
+	}
+	tests := []struct {
+		name        string
+		args        args
+		wantEnvVars map[string]string
+		wantErrs    []error
+		wantWarns   []string
+	}{
+		{
+			"Valid env variables",
+			args{[]string{
+				"DISABLE_UPDATES=True",
+				"DISABLE_TELEMETRY=False",
+				"METRICS_RESOLUTION=5m",
+				"METRICS_RESOLUTION_HR=5s",
+				"METRICS_RESOLUTION_LR=1h",
+				"DATA_RETENTION=72h",
+			}},
+			map[string]string{
+				"DATA_RETENTION":        "72h",
+				"DISABLE_TELEMETRY":     "false",
+				"DISABLE_UPDATES":       "true",
+				"METRICS_RESOLUTION":    "5m",
+				"METRICS_RESOLUTION_HR": "5s",
+				"METRICS_RESOLUTION_LR": "1h",
+			},
+			nil,
+			nil,
+		},
+		{
+			"Unknown env variables",
+			args{[]string{"UNKNOWN_VAR=VAL", "ANOTHER_UNKNOWN_VAR=VAL"}},
+			map[string]string{"ANOTHER_UNKNOWN_VAR": "val", "UNKNOWN_VAR": "val"},
+			nil,
+			[]string{
+				`unknown environment variable "UNKNOWN_VAR=VAL"`,
+				`unknown environment variable "ANOTHER_UNKNOWN_VAR=VAL"`,
+			},
+		},
+		{
+			"Invalid env variables values",
+			args{[]string{
+				"DISABLE_UPDATES=5",
+				"DISABLE_TELEMETRY=X",
+				"METRICS_RESOLUTION=5f",
+				"METRICS_RESOLUTION_HR=s5",
+				"METRICS_RESOLUTION_LR=1hour",
+				"DATA_RETENTION=keep one week",
+			}},
+			map[string]string{
+				"DATA_RETENTION":        "keep one week",
```

Comment:
Why do we return these values if they are wrong?

Reply:
Good point 👍